lib/ansible/modules/network/netvisor/pn_port_config.py

```diff
@@ -0,0 +1,382 @@
+    description:
+      - physical port mirror only.
+    required: False
+    type: bool
+  pn_reflect:
+    description:
+      - physical port reflection.
+    required: False
+    type: bool
+  pn_jumbo:
+    description:
+      - jumbo frames on physical port.
+    required: False
+    type: bool
+  pn_egress_rate_limit:
+    description:
+      - max egress port data rate limit.
+    required: False
+    type: str
+  pn_eth_mode:
+    description:
+      - physical Ethernet mode.
+    required: False
+    choices: ['1000base-x', 'sgmii', 'disabled', 'GMII']
+  pn_fabric_guard:
+    description:
+      - Fabric guard contfiguration.
+    required: False
+    type: bool
+  pn_local_switching:
+    description:
+      - no-local-switching port cannot bridge traffic to
+        another no-local-switching port.
+    required: False
+    type: bool
+  pn_lacp_priority:
+    description:
+      - LACP priority from 1 to 65535.
+    required: False
+    type: str
+  pn_send_port:
+    description:
+      - send port.
+    required: False
+    type: str
+  pn_port_mac_address:
+    description:
+      - physical port MAC Address.
+    required: False
+    type: str
+  pn_defer_bringup:
+    description:
+      - defer port bringup.
+    required: False
+    type: bool
+"""
+
+EXAMPLES = """
+- name: port config modify
+  pn_port_config:
+    pn_cliswitch: "sw01"
+    state: "update"
+    pn_port: "all"
+    pn_dscp_map: "foo"
+
+- name: port config modify
+  pn_port_config:
+    pn_cliswitch: "sw01"
+    state: "update"
+    pn_port: "all"
+    pn_host_enable: True
+"""
+
+RETURN = """
+command:
+  description: the CLI command run on the target node.
+  returned: always
+  type: str
+stdout:
+  description: set of responses from the port-config command.
+  returned: always
+  type: list
+stderr:
+  description: set of error responses from the port-config command.
+  returned: on error
+  type: list
+changed:
+  description: indicates whether the CLI caused changes on the target.
+  returned: always
+  type: bool
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.netvisor.pn_nvos import pn_cli, run_cli, booleanArgs
+
+
+def check_cli(module, cli):
+    """
+    This method checks for idempotency using the dscp-map-show name command.
+    If a user with given name exists, return True else False.
+    :param module: The Ansible module to fetch input parameters
+    :param cli: The CLI string
+    """
+    name = module.params['pn_dscp_map']
+
+    cli += ' dscp-map-show name %s format name no-show-headers' % name
+    out = module.run_command(cli.split(), use_unsafe_shell=True)[1]
+
+    out = out.split()
+
+    return True if name in out else False
+
+
+def main():
+    """ This section is for arguments parsing """
+
+    state_map = dict(
+        update='port-config-modify'
+    )
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            pn_cliswitch=dict(required=False, type='str'),
```

Comment:
`required=False` is the default value and not required to be passed explicitly. Applies to other places as well.